third_party/libunwindstack/tests/PeCoffFake.cpp

```diff
@@ -0,0 +1,322 @@
+
+  if constexpr (sizeof(typename PeCoffInterfaceType::AddressType) == 4) {
+    // Data offset only exists in 32-bit PE/COFF.
+    offset = SetData32(offset, 0);
+  }
+
+  // image_base
+  offset = SetMax64(offset, kLoadBiasFake, sizeof(typename PeCoffInterfaceType::AddressType));
+
+  offset = SetData32(offset, 0x1000);  // sect_alignment
+  offset = SetData32(offset, 0x200);   // file_alignment
+  offset = SetData16(offset, 0);       // major_os_system_version
+  offset = SetData16(offset, 0);       // minor_os_system_version
+  offset = SetData16(offset, 0);       // major_image_version
+  offset = SetData16(offset, 0);       // minor_image_version
+  offset = SetData16(offset, 0);       // major_subsystem_version
+  offset = SetData16(offset, 0);       // minor_subsystem_version
+  offset = SetData32(offset, 0);       // reserved1
+  offset = SetData32(offset, 0);       // image_size
+  offset = SetData32(offset, 0);       // header_size
+  offset = SetData32(offset, 0);       // checksum
+  offset = SetData16(offset, 0);       // subsystem
+  offset = SetData16(offset, 0);       // dll_flags
+
+  // stack_reserve_size
+  offset = SetMax64(offset, 0, sizeof(typename PeCoffInterfaceType::AddressType));
+  // stack_commit_size
+  offset = SetMax64(offset, 0, sizeof(typename PeCoffInterfaceType::AddressType));
+  // heap_reserve_size
+  offset = SetMax64(offset, 0, sizeof(typename PeCoffInterfaceType::AddressType));
+  // heap_commit_size
+  offset = SetMax64(offset, 0, sizeof(typename PeCoffInterfaceType::AddressType));
+
+  offset = SetData32(offset, 0);  // loader_flags
+
+  optional_header_num_data_dirs_offset_ = offset;
+  constexpr uint32_t kNumDirDataEntries = 10;
+  offset = SetData32(offset, kNumDirDataEntries);  // num_dir_data_entries
+
+  for (uint32_t i = 0; i < kNumDirDataEntries; ++i) {
+    offset = SetData32(offset, 0);
+    offset = SetData32(offset, 0);
+  }
+
+  SetData16(optional_header_size_offset_, offset - optional_header_start_offset_);
+
+  return offset;
+}
+
+template <typename PeCoffInterfaceType>
+uint64_t PeCoffFake<PeCoffInterfaceType>::SetSectionHeaderAtOffset(uint64_t offset,
+                                                                   std::string section_name,
+                                                                   uint64_t vmsize, uint64_t vmaddr,
+                                                                   uint64_t size,
+                                                                   uint64_t file_offset) {
+  CHECK(memory_);
+  std::string name_in_header = section_name;
+  if (section_name.size() > kSectionNameInHeaderSize) {
+    const uint64_t previous_offset =
+        section_names_in_string_table_.empty() ? 0 : section_names_in_string_table_.back().first;
+    const uint64_t previous_size = section_names_in_string_table_.empty()
+                                       ? 0
+                                       : section_names_in_string_table_.back().second.size() + 1;
```

Comment:
Could you add some parentheses here? To be clear about the `+1`?

Reply:
Good suggestion, done.

Reply:
Also, added a comment why the +1 is needed, as it's not apparent from the code here only (the strings are written to the string table somewhere else).